v2/concordium/types.proto

```diff
@@ -0,0 +1,459 @@
+  BakerAggregationVerifyKey aggregation_key = 4;
+}
+
+// Pending change to the stake either of a baker or delegator.
+message StakePendingChange {
+  message Reduce {
+    Amount new_stake = 1;
+    // Timestamp in milliseconds when the change takes effect.
+    uint64 effective_time = 2;
+  }
+  
+  oneof change {
+    Reduce reduce = 1;
+    // Remove the stake. The value is a timestamp of the effective time in
+    // milliseconds.
+    uint64 remove = 2;
+  }
+}
+
+// Information about how open the pool is to new delegators.
+enum OpenStatus {
+  OPEN_STATUS_OPENFORALL = 0;
+  OPEN_STATUS_CLOSEDFORNEW = 1;
+  OPEN_STATUS_CLOSEDFORALL = 2;
+}
+
+// A fraction of an amount with a precision of `1/100_000`.
+message AmountFraction {
+  uint32 parts_per_hundred_thousand = 1;
+}
+
+// Distribution of the rewards for the particular pool.
+message CommissionRates {
+  // Fraction of finalization rewards charged by the pool owner.
+  AmountFraction finalization = 1;
+  // Fraction of baking rewards charged by the pool owner.
+  AmountFraction baking = 2;
+  // Fraction of transaction rewards charged by the pool owner.
+  AmountFraction transaction = 3;
+}
+
+/// Additional information about a baking pool.
+/// This information is added with the introduction of delegation.
+message BakerPoolInfo {
+  // Whether the pool allows delegators.
+  OpenStatus open_status = 1;
+  // The URL that links to the metadata about the pool.
+  string url = 2;
+  // The commission rates charged by the pool owner.
+  CommissionRates commission_rates = 3;
+}
+
+// Information about the account stake, if the account is either a baker or a
+// delegator.
+message AccountStakingInfo {
+  message Baker {
+    Amount staked_amount = 1;
+    bool restake_earnings = 2;
+    BakerInfo baker_info = 3;
+    // If present, any pending change to the delegated stake.
+    optional StakePendingChange pending_change = 4;
+    // Present if the account is currently a baker, i.e., it is in the baking
+    // committee of the current epoch.
+    optional BakerPoolInfo pool_info = 5;
+  }
+
+  message Delegate {
+    // The amount that the account delegates.
+    Amount staked_amount = 1;
+    // Whether the earnings are automatically added to the staked amount.
+    bool restake_earnings = 2;
+    // The entity to which the account delegates.
+    DelegationTarget target = 3;
+    // If present, any pending change to the delegated stake.
+    optional StakePendingChange pending_change = 4;
+  }
+
+  oneof staking_info {
+    // The account is a baker.
+    Baker baker = 1;
+    // The account is a delegator.
+    Delegate delegate = 2;
+  }
+}
+
+// A sequence number that determines the ordering of transactions from the
+// account. The minimum sequence number is 1.
+message Nonce {
+  // The nonce
+  uint64 value = 1;
+}
+
+// An amount of microCCD.
+message Amount {
+  uint64 value = 1;
+}
+
+// The amount of microCCD.
+message CredentialIndex {
+  uint32 value = 1;
+}
```

Comment:
It's not used at present because map keys in protobuf can only be primitives. But it will be in some other responses. So I've updated the documentation.